IdentityCore/src/requests/sdk/msal/MSIDDefaultBrokerResponseHandler.m

```diff
@@ -121,21 +136,31 @@ - (MSIDBrokerResponse *)brokerResponseFromEncryptedQueryParams:(NSDictionary *)e
         }
     }
     
-    // Successful case
-    if ([NSString msidIsStringNilOrBlank:decryptedResponse[@"broker_error_domain"]]
-        && [decryptedResponse[@"success"] boolValue])
-    {
-        return [[MSIDAADV2BrokerResponse alloc] initWithDictionary:decryptedResponse error:error];
-    }
-    
-    // Failure case
     MSIDAADV2BrokerResponse *brokerResponse = [[MSIDAADV2BrokerResponse alloc] initWithDictionary:decryptedResponse error:error];
     
+    // Error when initializing broker response
     if (!brokerResponse)
     {
         return nil;
     }
     
+    // Successful response from broker
+    if ([NSString msidIsStringNilOrBlank:decryptedResponse[@"broker_error_domain"]]
```

Comment:
Had an off-line discussion with Rohit and Hieu. Agreed to see if there's a common place where this can be added. Additionally, we need to pass bearer as the scheme when saving additional tokens.

Reply:
Addressed. Update in the next commit following Rohit and Olga's comment. The check will happen in MSIDTokenResponseValidator